src/kontrol/foundry.py

```diff
@@ -375,13 +368,19 @@ def help_info() -> list[str]:
         )
         return res_lines
 
-    def proofs_with_test(self, test: str) -> list[Proof]:
-        proofs = [
-            self.get_optional_proof(pid)
-            for pid in listdir(self.proofs_dir)
-            if re.search(single(self._escape_brackets([test])), pid.split(':')[0])
+    def proof_ids_with_test(self, test: str, version: int | None = None) -> list[str]:
+        regex = single(self._escape_brackets([test]))
+        all_proof_ids: list[tuple[str, str, int]] = []
+        for pid in listdir(self.proofs_dir):
+            proof_type = pid.split('%')[0]
```

Comment:
I'm not sure this works. I think the `%` notation is not meant to demarcate a proof type, but rather as a separator that replaces the `/` in file path where the solidity file the function is found in relative to the foundry dir, so if a test is in `src/tests/integration/test-data/foundry/test/a/b/c/MyTest.t.sol` it would be something like `test%a%b%c%MyTest.test_function():0`

Reply:
Ahhhhh, I see. So I guess we should basically take all but the last segment for this one.

Reply:
Done.